config/base/src/lib.rs

```diff
@@ -256,3 +271,59 @@ pub mod view {
         pub const IS_HAS_VIEW: bool = true;
     }
 }
+
+pub mod proxy {
+    //! Module for configuration proxies' traits
+
+    use super::*;
+
+    /// Pseudo-default trait only used for doc generation
+    pub trait DocsDefault {
+        fn default() -> Self;
+    }
+
+    /// Trait used to convert configs from file and env
+    pub trait Combine: Sized + Serialize + DeserializeOwned {
+        /// Which type of [`Configuration`] it builds into
+        type Target;
+
+        /// Build the config, do the necessary checks
+        fn build(self) -> Result<Self::Target, derive::Error>;
+
+        /// If any of the fields in [`other`] are filled, they
+        /// override the values of the fields in [`self`].
+        fn combine(self, other: Self) -> eyre::Result<Self, eyre::Error>;
+
+        /// Construct [`Self`] from a path-like object.
+        ///
+        /// # Errors
+        /// - File not found.
+        /// - File found, but peer configuration parsing failed.
+        /// - The length of the array in raw JSON representation is different
+        /// from the length of the array in
+        /// [`self.sumeragi.trusted_peers.peers`], most likely due to two
+        /// (or more) peers having the same public key.
+        fn from_path<P: AsRef<Path> + Debug + Clone>(path: P) -> eyre::Result<Self, eyre::Error> {
```

Comment:
Don't know how robust it is to try to make a default `from_path` method for all proxy structs, but there definitely is some space for improvement in existing code, as we have 3 or 4 implementations of almost the same function scattered as `impl Configuration` here and there.

Reply:
Should be a separate trait. I'd say `LoadFromDisk` and make it a little richer.